### PR TITLE
[DO-NOT-MERGE] Saves stealth addresses accordingly when created via gui

### PR DESCRIPTION
**Pull request by marpme**

### CODE REVIEW

File: src/script/interpreter.cpp [hunk 1]
```diff
@@ -181,9 +181,9 @@ bool static IsLowDERSignature(const valtype &vchSig, ScriptError* serror) {
     // If the S value is above the order of the curve divided by two, its
     // complement modulo the order could have been used instead, which is
     // one byte shorter when encoded correctly.
-    if (!CPubKey::CheckLowS(vchSigCopy)) {
+    /*if (!CPubKey::CheckLowS(vchSigCopy)) {
         return set_error(serror, SCRIPT_ERR_SIG_HIGH_S);
-    }
+    }*/
```
Comment (marpme) [inline src/script/interpreter.cpp]: should definitely be reactivated before merge. Fix internals.

File: src/script/standard.cpp [hunk 2]
```diff
@@ -250,8 +251,10 @@ class CScriptVisitor : public boost::static_visitor<bool>
 {
 private:
     CScript *script;
+    bool shouldStealth;
 public:
     explicit CScriptVisitor(CScript *scriptin) { script = scriptin; }
+    CScriptVisitor(CScript *scriptin, bool shouldStealthIn) { script = scriptin; shouldStealth = shouldStealthIn; }
```
Comment (marpme) [inline src/script/standard.cpp]: Remove useless boolean

File: src/wallet/wallet.cpp [hunk 3]
```diff
@@ -3202,8 +3203,10 @@ bool CWallet::AddStealthAddress(CStealthAddress& sxAddr)
     encrypted_batch = new WalletBatch(*database);
     bool rv = encrypted_batch->WriteStealthAddress(sxAddr);
     
-    if (rv)
+    if (rv){
         SetAddressBook(sxAddr, sxAddr.label, "");
+        LogPrintf("Wrote %s Address to our address store.", sxAddr.label);
```
Comment (marpme) [inline src/wallet/wallet.cpp]: Unnecessary log here : )

**Comment by SwenVanZanten** [comment]
Can't send tx from UI

Schermafbeelding 2019-03-14 om 16 27 59

💥

**Comment by SwenVanZanten** [comment]
Can't send tx from UI

Schermafbeelding 2019-03-14 om 16 27 59

💥

File: src/qt/walletmodel.cpp [hunk 4]
```diff
@@ -177,9 +178,37 @@ WalletModel::SendCoinsReturn WalletModel::prepareTransaction(WalletModelTransact
             setAddress.insert(rcp.address);
             ++nAddresses;
 
-            CScript scriptPubKey = GetScriptForDestination(DecodeDestination(rcp.address.toStdString()));
-            CRecipient recipient = {scriptPubKey, rcp.amount, rcp.fSubtractFeeFromAmount};
-            vecSend.push_back(recipient);
+            CTxDestination dest = DecodeDestination(rcp.address.toStdString());
+            if(dest.type() == typeid(CStealthAddress)){
```
Comment (marpme) [inline src/qt/walletmodel.cpp]: default fallback missing

**Pull request merged**